### DISCUSSION TITLE
Reload configuration?

### DISCUSSION THREAD

**Discussion by xnoreq** [DNSCrypt/dnscrypt-proxy#2161] [Potential issues]
Hello,
how can I reload the configuration (including allow/block lists)?

Currently, it only seems possible to restart dnscrypt-proxy, which will destroy its cache. There also doesn't seem to be something like unbound-control dump_cache and load_cache, which can be used to make the cache survive a restart.

**Comment by lifenjoiner** [comment]
To me, reloading cache is not important for a local DNS.

**Comment by xnoreq** [comment]
I do not want to reload the cache but the configuration. 

Blocklists need to be updated regularly.
Currently with dnscrypt-proxy this means the service needs to be restarted regularly as well.
Not only does this destroy the cache, it also means there is a downtime.

**Comment by Atalantia** [comment]
To reset you have to delete:
public-resolvers.md
public-resolvers.md.minisig
relays.md
relays.md.minisig

Configure dnscrypt-proxy is a pain int the **** and there aren't any good tutorials around. I managed only the default, out of the box to work. Any modification isn't possible (for me).

**Comment by lifenjoiner** [comment]
Hot-reloading of configuration files has been implemented since [v2.1.10](https://github.com/DNSCrypt/dnscrypt-proxy/releases/tag/2.1.10)!

Too many people don't read either the [readme/wiki](https://github.com/DNSCrypt/dnscrypt-proxy?tab=readme-ov-file#overview) or the release [descriptions](https://github.com/DNSCrypt/dnscrypt-proxy/releases/)/[changelog](https://github.com/DNSCrypt/dnscrypt-proxy/blob/master/ChangeLog).
